src/Microsoft.Azure.WebJobs/ExtensionAttribute.cs

```diff
@@ -13,8 +13,8 @@ public sealed class ExtensionAttribute : Attribute
     {
         public ExtensionAttribute(string name, string configurationSection = null)
         {
-            Name = name;
-            ConfigurationSection = configurationSection ?? name;
+            Name = name ?? throw new ArgumentNullException(nameof(name));
+            ConfigurationSection = configurationSection;
```

Comment:
Modified this to allow configuration section to be `null`, as using the name would be problematic as it is a human readable name.

Reply:
FYI @mathewc